src/appleseed/renderer/modeling/aov/pixelsamplecountaov.cpp

```diff
@@ -72,90 +74,20 @@ const char* PixelSampleCountAOV::get_model() const
     return PixelSampleCountAOVModel;
 }
 
-namespace
-{
-    // Return the maximum sample/pixel count from `image`'s pixels inside of `crop_window`.
-    float get_max_spp_count(
-        const Image*    image,
-        const AABB2u&   crop_window)
-    {
-        float max_spp = 0.0f;
-
-        for (size_t y = crop_window.min.y; y <= crop_window.max.y; ++y)
-        {
-            for (size_t x = crop_window.min.x; x <= crop_window.max.x; ++x)
-            {
-                Color3f color;
-                image->get_pixel(x, y, color);
-                max_spp = max(color[0], max_spp);
-            }
-        }
-
-        return max_spp;
-    }
-
-    void fill_aov(
-        Image*          image,
-        const AABB2u&   crop_window,
-        const Color3f&  color)
-    {
-        for (size_t y = crop_window.min.y; y <= crop_window.max.y; ++y)
-        {
-            for (size_t x = crop_window.min.x; x <= crop_window.max.x; ++x)
-                image->set_pixel(x, y, color);
-        }
-    }
-}
-
 void PixelSampleCountAOV::post_process_image(const Frame& frame)
 {
-    static const Color3f Blue(0.0f, 0.0f, 1.0f);
-    static const Color3f Red(1.0f, 0.0f, 0.0f);
-
     const AABB2u& crop_window = frame.get_crop_window();
+    ColorMap color_map;
+    color_map.set_palette_from_array(InfernoColorMap, countof(InfernoColorMap) / 3);
+
+    float min_spp = static_cast<float>(m_min_spp);
+    float max_spp = 0.0f;
+    if (m_max_spp == 0)
+        color_map.find_min_max_red_channel(*m_image, crop_window, min_spp, max_spp);
+    else
+        max_spp = static_cast<float>(m_max_spp);
 
-    //
```

Comment:
Can you please keep this comment (and update it if needed)? It's quite important as it explains the logic behind the determination of max SPP.

Reply:
Sure :)